### PR TITLE
Added Term Entry HTML: Tags: `<source>`

### LINKED ISSUE #528: [Term Entry] HMTL Tags : `<source>`
### Reason/inspiration (optional)

Write a term entry about the `<source>` attribute that includes: 

- Brief "Syntax" and "Example" sections.
- Info about `<source>` being an empty element and how it should never use a closing tag or slash `/`.
- A short breakdown of some of its common attributes such as:
  - `media`
  - `sizes`
  - `src`
  - `srcset`
  - `type`

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by ethan-master-coding**
### Description

Closes #528 and #558 (duplicate mistake - @Dusch4593)

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by ethan-master-coding** [comment]
@Dusch4593 How do I add this to the main page of the HTML tags? Will, it auto do that.

**Comment by denisCabrera** [comment]
Hi,

I am relatively new to contributing. I am curious about the check that failed and where to move from here. Would love any feedback.

Thank you

**Comment by SSwiniarski** [comment]
@ethan-master-coding, I'll do the first review. FYI, running "yarn format" on this file should get rid of the check error on this. (It's due to some formatting it doesn't like.)

File: content/html/concepts/tags/terms/source/source.md
Comment (SSwiniarski) [inline content/html/concepts/tags/terms/source/source.md:8]:
```suggestion
  - 'Tags'
  - 'DOM'
  - 'Elements'
  - 'Semantic'
```

File: content/html/concepts/tags/terms/source/source.md
Comment (SSwiniarski) [inline content/html/concepts/tags/terms/source/source.md:25]:
```suggestion
The browser will first try and play the **./video.mp4** file and if it can't it will try the **./video.webm** file. And if neither of those work it will display "Video not supported". You can have as many `<source>` elements as you want as long as you have files for each one.
```

File: content/html/concepts/tags/terms/source/source.md
Comment (SSwiniarski) [inline content/html/concepts/tags/terms/source/source.md:14]:
```suggestion
The `<source>` element allows for multiple sources to be in one element in case one type is not supported. The `<source>` element is nested within an `<video>`, `<img>` or another tag that embeds content. It allows for multiple files to be referenced in one element in case a file type is not supported. If one file type can not be displayed, the browser will attempt to display the next.
```

File: content/html/concepts/tags/terms/source/source.md
Comment (SSwiniarski) [inline content/html/concepts/tags/terms/source/source.md:15]:
```suggestion

## Example

```

File: content/html/concepts/tags/terms/source/source.md
Comment (SSwiniarski) [inline content/html/concepts/tags/terms/source/source.md:17]:
```suggestion
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@ethan-master-coding, I've completed my review and left a few suggestions.

**Comment by ethan-master-coding** [comment]
@SSwiniarski Finished going through your suggestions. I accepted them.

File: content/html/concepts/tags/terms/source/source.md
Comment (Dusch4593) [inline content/html/concepts/tags/terms/source/source.md:29]:
```suggestion
The browser will first try and play the **./video.mp4** file and if it can't it will try the **./video.webm** file. And if neither of those work it will display "Video not supported".  A large amount of `<source>` elements can be used as long as there is a valid file sourced for each one.
```

For Docs entries, we want to avoid using 1st- or 2nd-person pronouns and opt for a friendly, but manual voice that uses 3rd-person sparingly.

**Comment by ethan-master-coding** [comment]
All edits look good. I accepted the changes. Also can you look at my new PR #563

**Comment by Dusch4593** [comment]
@ethan-master-coding I made a change in a6b71fb83307602f69c1eb32d0834898ff1d161d where I took "Semantic" from the `Tags` metadata because I'm not 100% positive that the `<source>` tag is functionally semantic in the way that `<main>` or `<article>` would be.

**Comment by Dusch4593** [comment]
> @Dusch4593 All edits look good. I accepted the changes. Also can you look at my new PR #563

Will look at when I have a sec. We've been getting a flux of PRs lately so turnarounds make take a bit.

**Comment by SSwiniarski** [comment]
> @ethan-master-coding I made a change in [a6b71fb](https://github.com/Codecademy/docs/commit/a6b71fb83307602f69c1eb32d0834898ff1d161d) where I took "Semantic" from the `Tags` metadata because I'm not 100% positive that the `<source>` tag is functionally semantic in the way that `<main>` or `<article>` would be.

@Dusch4593, FYI, that was my addition. I'll defer to you.

**Review by SSwiniarski — APPROVED** [review]

**Comment by SSwiniarski** [comment]
@ethan-master-coding,  Ready for review two.

File: content/html/concepts/tags/terms/source/source.md
Comment (Dusch4593) [inline content/html/concepts/tags/terms/source/source.md:28]:
```suggestion
The browser will first try and play the **./video.mp4** file and if it can't it will try the **./video.webm** file. If neither of those work one of the following will happen:

* `<video>` will remain inline within the page's markup with only the controls visible (if they're enabled).
* Fallback content (like "Video not supported") is displayed for browsers that do not support HTML5, which includes the `<audio>` or `<video>` tags.

A large amount of `<source>` elements can be used as long as there is a valid file sourced for each one.
```

File: content/html/concepts/tags/terms/source/source.md
Comment (Dusch4593) [inline content/html/concepts/tags/terms/source/source.md:21]:
```suggestion
<video controls>
```

This will allow the empty video screen to be visible on most browsers.

File: content/html/concepts/tags/terms/source/source.md
Comment (Dusch4593) [inline content/html/concepts/tags/terms/source/source.md:16]:
```suggestion
The `<source>` element allows for multiple sources or file formats to be associated with the `<audio>`, `<video>`, and `<picture>` tags. In case one source is incompatible with a given browser, another one can be referenced. If all else fails, the browser will attempt to display fallback content.
```

Comment (Dusch4593) [reply content/html/concepts/tags/terms/source/source.md:16]: There's no compatibility between the `` and `<source>` elements, AFAIK. The `` tag has a `src` attribute, after all. 😄

File: content/html/concepts/tags/terms/source/source.md
Comment (Dusch4593) [inline content/html/concepts/tags/terms/source/source.md:17]: @ethan-master-coding Let's make sure to add a brief "Syntax" section here above the "Example". This is a format that we'd like to follow for term entries.

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@ethan-master-coding Left some additional comments/suggestions for you to look at. Tag us when you're finished and we shall move onward! 😃

**Comment by ethan-master-coding** [comment]
@Dusch4593 Finished going through your suggestions.

**Review by Dusch4593 — APPROVED** [review]
@ethan-master-coding Nice! Ready for next review 👍 

Side-note: I'm curious, when you seem to make changes locally, does your terminal/command line look like this after you commit (and before you push)?

Screen Shot 2022-02-25 at 10 55 01 PM

This is in regards to the checks sometimes not passing when we push here.

**Comment by ethan-master-coding** [comment]
@Dusch4593 The changes today were all from within the browser. And when I am writing a new article it's in vs code with gh desktop

**Comment by Dusch4593** [comment]
> @Dusch4593 The changes today were all from within the browser. And when I am writing a new article it's in vs code with gh desktop

NIce! Was just curious, is all. I think our lint/format script doesn't pick up changes coming from non-local places 😎.

**Comment by ethan-master-coding** [comment]
@Dusch4593 If I were to make articles for Phaser.js would it be its own folder or within JS?

File: content/html/concepts/tags/terms/source/source.md
Comment (KTom101) [inline content/html/concepts/tags/terms/source/source.md:18]:
```suggestion

Attributes provide additional characteristics to the element. Some common attributes used with `<source>` are:
  - `src`: specifies the URL of the media, used for `<audio>` and `<video>` tags
  - `srcset`: defines a set of images for the browser to choose from, used with `<picture>`  
  - `media`: tells which device the document is optimized for
  - `sizes`: sets out the size of visual media icons
  - `type`: specifies the content type e.g., `video/mp4` or `audio/ogg`
	  
## Example
```

File: content/html/concepts/tags/terms/source/source.md
Comment (KTom101) [reply content/html/concepts/tags/terms/source/source.md:17]: Add backticks to end the code block. I can't do this in the comment because it interferes with the suggestion block.

**Review by KTom101 — COMMENTED** [review]
@ethan-master-coding Here's another review.

cc: @Dusch4593

**Comment by Dusch4593** [comment]
> @Dusch4593 If I were to make articles for Phaser.js would it be its own folder or within JS?

I would think it would go in the JS topic as a concept entry: `javascript/concepts/phaser-js/phaser-js.md`

**Comment by ethan-master-coding** [comment]
@KTom101 I accepted those changes and made the changes you requested.

**Comment by ethan-master-coding** [comment]
I think we should hold off on merging until the `<picture>` tag entry gets merged (#545) so we can link to it.

**Comment by Dusch4593** [comment]
> I think we should hold off on merging until the `<picture>` tag entry gets merged (#545) so we can link to it.

@ethan-master-coding Was thinking the opposite thing for this entry. 😃 Both this PR and that one are really close to being ready for merge, so if we need to edit to add the links that won't be a problem.

**Comment by ethan-master-coding** [comment]
> > I think we should hold off on merging until the `<picture>` tag entry gets merged (#545) so we can link to it.
> 
> @ethan-master-coding Was thinking the opposite thing for this entry. 😃 Both this PR and that one are really close to being ready for merge, so if we need to edit to add the links that won't be a problem.

Guess I will get an extra contribution for very little work then. 😂

File: content/html/concepts/tags/terms/source/source.md
Comment (KTom101) [inline content/html/concepts/tags/terms/source/source.md:20]:
```suggestion
`<source>` is an empty tag and has no children or closing tags.
```

File: content/html/concepts/tags/terms/source/source.md
Comment (KTom101) [inline content/html/concepts/tags/terms/source/source.md:31]:
```suggestion
  - `type`: specifies the content type e.g., `video/mp4` or `audio/ogg`, can be used with any of the three parent elements of `<source>` (`<picture>`, `<video>`, or `<audio>`).
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@ethan-master-coding @Dusch4593 and I took another look at this and we have further suggestions to improve this entry even more. This is a review with our comments combined.

**Comment by ethan-master-coding** [comment]
All edits look good. I accepted the changes.

**Review by KTom101 — APPROVED** [review]
@ethan-master-coding Great! If everyone is cool with this, then we can get this merged!
cc @Dusch4593 @SSwiniarski @yangc95

**Comment by Dusch4593** [comment]
@ethan-master-coding This looks ready to merge! 🚀 

Made a slight change in 6fd0fe2274dc9ef1fc8c8d25e6f2fe8a3d89ff9c where I linked `<picture>` to the entry that is not yet live but will be very soon after this one. Thank you for your work on this and look forward to working with you on the next one (P.S. I just assigned you to #569)

**Comment by Dusch4593** [comment]
When the entry is live, it will be here: https://www.codecademy.com/resources/docs/html/tags/source

**Comment by ethan-master-coding** [comment]
@Dusch4593 when will it go live?

**Pull request merged**
